enumerable_methods.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while self[i]
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_each { |e| yield(index(e), e) }
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_items = []
+    my_each { |item| selected_items << item if yield(item) }
+    selected_items
+  end
+
+  def my_all?
+    my_each { |item| return false unless yield(item) }
+    true
+  end
+
+  def my_any?
+    my_each { |item| return true if yield(item) }
+    false
+  end
```

Comment:
What happens when no block is given? From your implementation, you always return `false`. So if I run `[nil, true].my_any?` I would get `false`. You need to fix this.

Consider having another execution path for when a block is not given.

Also, when I run `[nil, true, 2].my_any?(Integer)`, I get the error below:

> ArgumentError:
>        wrong number of arguments (given 1, expected 0)